### ISSUE TITLE
v1.7 test fails t0010-generate-tests.sh on OS X

### ISSUE THREAD

**Issue by HacKanCuBa** [HacKanCuBa/passh#8]
**From the pass mailing list:**

Hi all

I downloaded the version 1.7, and `make test` fails:
```
$ make test
ok 1 - Make sure we can run pass
ok 2 - Make sure we can initialize our test store
# passed all 2 test(s)
1..2
ok 1 - Test "generate" command
not ok 2 - Test replacement of first line
#
# "$PASS" insert -m cred2 <<<"$(printf "this is a big\\npassword\\nwith\\nmany\\nlines\\nin it bla bla")" &&
# PATH="$TEST_HOME:$PATH" FAKE_PWGEN_PASSWORD="This is a fake password" "$PASS" generate -i cred2 23 &&
# [[ $("$PASS" show cred2) == "$(printf "This is a fake password\\npassword\\nwith\\nmany\\nlines\\nin it bla bla")" ]]
#
# failed 1 among 2 test(s)
1..2
make: *** [tests/t0010-generate-tests.sh] Error 1
```

in details:

```
$ tests/t0010-generate-tests.sh -v
expecting success:
"$PASS" init $KEY1 &&
"$PASS" generate cred 19 &&
[[ $("$PASS" show cred | wc -m) -eq 20 ]]

Password store initialized for CF90C77B
The generated password for cred is:
5NwpOZ}kyR>Q6%}A*W8
ok 1 - Test "generate" command

expecting success:
"$PASS" insert -m cred2 <<<"$(printf "this is a big\\npassword\\nwith\\nmany\\nlines\\nin it bla bla")" &&
PATH="$TEST_HOME:$PATH" FAKE_PWGEN_PASSWORD="This is a fake password" "$PASS" generate -i cred2 23 &&
[[ $("$PASS" show cred2) == "$(printf "This is a fake password\\npassword\\nwith\\nmany\\nlines\\nin it bla bla")" ]]

Enter contents of cred2 and press Ctrl+D when finished:

Usage: password-store.sh generate [--no-symbols,-n] [--clip,-c] [--qrcode,-q] [--in-place,-i | --force,-f] pass-name [pass-length]
not ok 2 - Test replacement of first line
#
# "$PASS" insert -m cred2 <<<"$(printf "this is a big\\npassword\\nwith\\nmany\\nlines\\nin it bla bla")" &&
# PATH="$TEST_HOME:$PATH" FAKE_PWGEN_PASSWORD="This is a fake password" "$PASS" generate -i cred2 23 &&
# [[ $("$PASS" show cred2) == "$(printf "This is a fake password\\npassword\\nwith\\nmany\\nlines\\nin it bla bla")" ]]
#

# failed 1 among 2 test(s)
1..2
```

Same error both on OS X El Capitan, and macOS Sierra.

Best regards, Isidoro

_______________________________________________
Password-Store mailing list
Password-Store@lists.zx2c4.com
https://lists.zx2c4.com/mailman/listinfo/password-store

**Comment by HacKanCuBa** [comment]
I had to install bash 4.x for my user on macOS to fix pass 1.7 auto-completion errors. macOS was using bash 3.x by default.
-- James Wald <james.wald@gmail.com>

**Comment by HacKanCuBa** [comment]
I also PM'ed Jason about this issue this morning.

The reason is: t0010-generate-tests.sh (also t0200-edit-tests.sh)
overload tr(1) to force the generation of a special password.

Unfortunately, this does not only affect the intended line of code
though, but also the loading of the platform.sh.

Cheers
David
-- Dahlberg, David <david.dahlberg@fkie.fraunhofer.de>
